### PR TITLE
JSON output for hq job list and hq submit

### LINKED ISSUE #13: Switch to the new JSON output for parsing the output of HQ
See e.g. docs here:
https://it4innovations.github.io/hyperqueue/latest/cli/output-mode/#json

and PR with comments here: https://github.com/It4innovations/hyperqueue/pull/270

I think it should be used at least for `hq jobs` and `hq submit` (and possibly `hq delete` if it is also supported there?)

### PR DESCRIPTION
**Pull request by unkcpz**
fixes #13 

Instead of using regex to parse the command output, HQ provide output mode JSON which we can get output to a dict to avoid using regex.

### CODE REVIEW

**Comment by unkcpz** [comment]
@t-reents You may want to take a look and continue on https://github.com/aiidateam/aiida-hyperqueue/pull/21?

**Comment by t-reents** [comment]
Hi @unkcpz 
You mean to also implement the `parse_output` method which builds on the `detailed_job_info_command` I implemented? I'm not 100% sure what you have in mind

**Comment by unkcpz** [comment]
Hi @t-reents, I just want to pin you as reviewer but you were not in the org so I am not able to make you the reviewer. BTW, this PR contains some change in test, it may help you with adding test to #21.

File: aiida_hyperqueue/scheduler.py [hunk 1]
```diff
@@ -222,24 +217,30 @@ def _parse_joblist_output(self, retval: int, stdout: str, stderr: str) -> list:
...
-                job_info.title = job_dict["name"]
-                job_info.job_state = _MAP_STATUS_HYPERQUEUE[job_dict["state"].upper()]
-                # TODO: In principle more detailed information can be parsed for each job by `hq job info`, such as cpu, wall_time etc.
-                job_info_list.append(job_info)
+        job_info_list = []
+        for hq_job_dict in hq_job_info_list:
+            job_info = JobInfo()
+            job_info.job_id = hq_job_dict["id"]
+            job_info.title = hq_job_dict["name"]
+            stats: t.List[str] = [
```
Comment (t-reents) [inline aiida_hyperqueue/scheduler.py:233]: Mainly out of curiosity, why do you explicitly add a type hint here? I'm mainly thinking about consistency, as I didn't see it anywhere else in the plugin apart from the function definitions?

**Review by t-reents — APPROVED** [review]
Just a small comment, apart from that it's fine

File: aiida_hyperqueue/scheduler.py (hunk 1 above)
Comment (unkcpz) [reply aiida_hyperqueue/scheduler.py:233]: Just to make my IDE happy, I am using neovim with ruff check, which is quite strict on typing. The type of elements in the list is deducted from the key of `hq_job_dict["task_stats"]`.
By tell my LSP the type of element explicitly, `stats[0]` knows it is a str and help me to hint `upper()` is a valid method to call.

**Pull request merged**